Scripts/RemediationScripts/Remediate-TransparentDataEncryptionForSQLDatabases.ps1

```diff
@@ -514,118 +611,130 @@ function Disable-TransparentDataEncryptionForSQLDatabases
+            $skippedSQLServerDatabases = @()
+        
+            if ($_.IsSynapseWorkspace -eq $false)
             {
-                $sqlServersRolledBack += $databaseDetail
+                $databaseList | ForEach-Object{
+                    $databaseName = $_
+                    $tdeStatus = $(Set-AzSqlDatabaseTransparentDataEncryption -ResourceGroupName $resourceGroupName -ServerName $serverName -DatabaseName $databaseName -State 'Disabled').State
+                    if ($tdeStatus -eq 'Disabled')
+                    {
+                        $rolledbackSQLServerDatabases += $databaseName
+                    }
+                    else
+                    {
+                        $skippedSQLServerDatabases += $databaseName
+                    }
+                }
+
+                $rollbackSummary += $databaseDetail | Select-Object @{N='ServerName';E={$serverName}},
+                                                                   @{N='ResourceGroupName';E={$resourceGroupName}},
+                                                                   @{N='RolledbackSQLServerDatabases';E={$rolledbackSQLServerDatabases}},
+                                                                   @{N='SkippedSQLServerDatabases';E={$skippedSQLServerDatabases}}
             }
             else
             {
-                $skippedSQLServerDatabases += $databaseDetail
-            }
-        }
-        catch
-        {
-            $skippedSQLServerDatabases += $databaseDetail
-            Write-Host "Error occurred while disabling Transparent Data Encryption (TDE) for SQL Server database: Resource Group Name - $($resourceGroupName), Server Name - $($serverName), Database Name - $($databaseName). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
-            Write-Host "Skipping this database. Transparent Data Encryption (TDE) will not be disabled for this database."
-        }
-    }
+                $databaseList | ForEach-Object{
+                    $sqlPoolName = $_
+                    $tdeStatus = $(Set-AzSynapseSqlPoolTransparentDataEncryption -ResourceGroupName $resourceGroupName -WorkspaceName $serverName -Name $sqlPoolName -State 'Disabled').State
+                    if ($tdeStatus -eq 'Disabled')
+                    {
+                        $rolledbackSQLServerDatabases += $sqlPoolName
+                    }
+                    else
+                    {
+                        $skippedSQLServerDatabases += $sqlPoolName
+                    }
+                }
 
-    if (($skippedSQLServerDatabases | Measure-Object).Count -eq 0)
-    {
-        Write-Host "Transparent Data Encryption (TDE) successfully disabled for all $($totalSQLServerDatabases) SQL Server database(s)." -ForegroundColor $([Constants]::MessageType.Update)
+                $rollbackSummary += $databaseDetail | Select-Object @{N='ServerName';E={$serverName}},
```

Comment:
Can this be combined with the corresponding block starting line 689?

Reply:
Done.